cartographer_grpc/mapping_server.cc

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "cartographer_grpc/mapping_server.h"
+
+#include "cartographer_grpc/proto/cartographer_service.grpc.pb.h"
+#include "glog/logging.h"
+
+namespace cartographer_grpc {
+
+MappingServer::MappingServer(
+    const proto::MappingServerOptions& mapping_server_options)
+    : map_builder_(mapping_server_options.map_builder_options()) {
+  framework::Server::Builder server_builder;
+  server_builder.SetServerAddress(mapping_server_options.server_address());
+  server_builder.SetNumberOfThreads(mapping_server_options.num_grpc_threads());
+  server_ = server_builder.Build();
+  server_->SetExecutionContext(
+      cartographer::common::make_unique<SlamExecutionContext>(&map_builder_));
```

Comment:
The unique_ptr gets ownership of this member variable, I believe it will be deleted twice. Fix this (by creating the map_builder in this constructor with new).

Reply:
As discussed the arguments in the parenthesis are passed to `SlamExecutionContext`'s constructor